great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -81,6 +83,33 @@ def _splitter_method_exists(cls, v: str):
             raise ValueError(f"unexpected value; permitted: '{permitted_values_str}'")
         return v
 
+    def test_connection(self, table_asset: TableAsset) -> None:
+        """Test the connection for the ColumnSplitter.
+
+        Args:
+            table_asset: The TableAsset to which the ColumnSplitter will be applied.
+
+        Raises:
+            TestConnectionError: If the connection test fails.
+        """
+        # A TypeVar for Datasource would get rid of this assertion,
+        # but circularity between Datasource <-> DataAsset is an issue.
+        assert isinstance(table_asset.datasource, SQLDatasource)
+        engine: sqlalchemy.engine.Engine = table_asset.datasource.get_engine()
+        inspector: sqlalchemy.engine.Inspector = sqlalchemy.inspect(engine)
+
+        columns: list[dict[str, Any]] = inspector.get_columns(
```

Comment:
@NathanFarmer Observation: Inspect will not always work.  While I am not certain of the best path forward, perhaps considering our fallback methods in metrics/util.py might help.  Thank you.

Reply:
Thanks @alexsherstinsky! What I was able to glean from `metrics/util.py` is that this might not work for `mssql` or `trino`. We will want to handle that logic when we implement subclasses such as `MSSQLDatasource` and `TrinoDatasource`. A good think to keep in mind.

Reply:
Maybe we don't want to test the connection on splitters by default. That is, since this works on postgres but not on generic splitters, maybe we want to not check the connection here but instead check the connection on the classes we know this works for.
That is, what type of errors would a `mssql` and `trino` user see here? Also, if this would work without for them if we didn't do this check, it seems like this check is too high up the hierarchy.

Reply:
Following up from our internal discussion, the forks seen in `metrics/util.py` may be due to temp table use by `SqlAlchemyExecutionEngine` and we will be adding testing for various dialects to ensure this works.